docs/general/Mainnet/high_level_architecture.md

```diff
@@ -0,0 +1,37 @@
+---
+sidebar_position: 1
+---
+
+# High-level architecture
+
+![architecture_movement.png](./images/architecture_movement.png)
+
+## Data Availability
+
+The DA service is used to store transaction data and make it available to the network. The service is accessed via [DA light nodes](./node_level_architecture.md#da-light-node).
+
+Celestia is used as the Data Availability (DA) layer.
+
+## Sequencing
+
+In order to protect the DA layer from spam, only certain [full nodes](./node_level_architecture.md#full-node) receive authorization to sequence transactions in ordered transaction-batches (called proto-blocks). These full nodes provide basic sequencing for transactions ingressed from the users. The full node then sends the proto-block to the DA service using the [DA light node](node_level_architecture.md#da-light-node).
+
+The DA service sequences the proto-blocks and provides the final ordering of transactions to the network in L2-blocks.
```

Comment:
```suggestion
The DA service sequences transaction batches (which have the form of data blobs). This results in proto-blocks, and where each proto-block may contain several transaction-batches. The proto-block is the final ordering step of transactions. Next stop: execution.

## Execution

The [full node](./node_level_architecture.md#full-node) receives proto-blocks via the [DA light node](node_level_architecture.md#da-light-node) and executes the transactions in the proto-blocks. The full node then creates L2 blocks from the proto-blocks, which contain additional information such as the new state root after applying the transactions.
```

Reply:
suggesting to add a section on execution, since there should be a step between 
1) the ordered (state-root free) data blob (i.e. protoBlock) that comes from the DA and 
2) the block that is created from this protoBlock. 

Do you think this is somewhat reflected in code?

Reply:
Yes, all is accurate.